kmip/services/server/engine.py

```diff
@@ -1279,3 +1288,63 @@ def _process_discover_versions(self, payload):
         )
 
         return response_payload
+
+    def _process_get_attribute_list(self, payload):
+        self._logger.info("Processing operation: Get Attribute List")
+
+        unique_identifier = self._id_placeholder
+        if payload.uid:
+            unique_identifier = payload.uid
+
+        object_type = self._get_object_type(unique_identifier)
+
+        managed_object = self._data_session.query(object_type).filter(
+            object_type.unique_identifier == unique_identifier
+        ).one()
+
+        response_payload = get_attribute_list.GetAttributeListResponsePayload(
+            uid=unique_identifier,
+            attribute_names=managed_object.get_attribute_list()
+        )
+
+        return response_payload
+
+    def _process_get_attributes(self, payload):
+        pass
```

Comment:
This should raise `exceptions.OperationNotSupported` with the appropriate message.